ViewControllers/StockFeedViewController.m

```diff
@@ -106,35 +105,46 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
         StockDetailsViewController *stockDetailsController = [segue destinationViewController];
         NSIndexPath *index = self.stockTableView.indexPathForSelectedRow;
         Stock *dataToPass = self.stocksArray[index.row];
-        if(self.isFiltered){
-            dataToPass = self.filteredStocksArray[index.row];
-        }
+        stockDetailsController.stock = dataToPass;
+    } else if ([segue.identifier isEqualToString: @"searchToDetails"]){
+        StockDetailsViewController *stockDetailsController = [segue destinationViewController];
+        NSIndexPath *index = self.searchTableView.indexPathForSelectedRow;
+        Stock *dataToPass = self.fetchedStockCache[index.row];
         stockDetailsController.stock = dataToPass;
     }
 }
 
 - (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
-    StockCell *stockCell = [tableView dequeueReusableCellWithIdentifier:@"stockCell"];
-    Stock *stock = self.stocksArray[indexPath.row];
-    if (self.isFiltered){
-        stock = self.filteredStocksArray[indexPath.row];
+    if (tableView == self.stockTableView){
+        StockCell *stockCell = [tableView dequeueReusableCellWithIdentifier:@"stockCell"];
+        Stock *stock = self.stocksArray[indexPath.row];
+        stockCell.stock = stock;
+        stockCell.selectionStyle = nil;
+        return stockCell;
```

Comment:
Seems like there could be some way to rearrange the logic/code here to not duplicate the last 3 lines of these conditional blocks